crates/uv-installer/src/plan.rs

```diff
@@ -97,17 +97,20 @@ impl<'a> Planner<'a> {
                     [] => {}
                     [installed] => {
                         let source = RequirementSource::from(dist);
-                        match RequirementSatisfaction::check(installed, &source)? {
-                            RequirementSatisfaction::Mismatch => {
+                        match RequirementSatisfaction::check(installed, &source) {
+                            Ok(RequirementSatisfaction::Mismatch) => {
                                 debug!("Requirement installed, but mismatched:\n  Installed: {installed:?}\n  Requested: {source:?}");
                             }
-                            RequirementSatisfaction::Satisfied => {
+                            Ok(RequirementSatisfaction::Satisfied) => {
                                 debug!("Requirement already installed: {installed}");
                                 continue;
                             }
-                            RequirementSatisfaction::OutOfDate => {
+                            Ok(RequirementSatisfaction::OutOfDate) => {
                                 debug!("Requirement installed, but not fresh: {installed}");
                             }
+                            Err(err) => {
+                                debug!("Failed to deserialize cached requirements for: {installed} ({err})");
```

Comment:
We might want this to be `warn!`

Reply:
I guess this can fail for other reasons though. Should it be more discriminate or no?

Reply:
At least in the current implementation, it actually is all CacheInfo calls that make this fallible. warn feels too aggressive for "oops the cache was invalid"?

Reply:
But that's probably out-of-scope.

Reply:
Should we move this into `RequirementSatisfaction::check`? If it's already non-fatal for other kinds of errors, maybe it shouldn't return `Result` at all and we should just ignore these errors within that method.

Reply:
That sounds like a good idea... hmm interesting SitePackages::satisfies_requirements also `?`s this method and it probably shouldn't..?